internal/events/aggregator.go

```diff
@@ -559,6 +587,13 @@ func (ag *aggregator) attemptMessageDispatch(ctx context.Context, msg *core.Mess
 		}
 	}
 
+	if valid && msg.Header.Group != nil {
```

Comment:
I was thinking hard about the implications of moving this check fully out of the DX callback at the point we _store_ the data+messages, until the point we do the aggregation to _confirm_ the messages.

The positive is very important, and clearly required: We handle the case that the group contains identities that need to be verified in order per the the blockchain, and the aggregation of the pin for those identities might happen _after_ the receipt.

The concern is that we're reducing the amount of verification we do before we store the data+messages.

I believe there is some part we can still verify in the DX callback, and we should do that:
- We know DX would have rejected the receipt of the message, unless it came from a `peer` that was already registered
- We know the `batch` when it was received contains a node UUID
- In the DX callback I believe then we should:
   - Look up the node in the batch payload by UUID
   - Verify that the peer of the node looked up by UUID, matches the DX Peer in the DX Callback
- Then for easier visibility into the history, on unconfirmed messages, I suggest:
   - Use the Node UUID (rather than `peer`) to be the thing we store against the batch
   - Add a field to `Message` such as `node` which contains a UUID:
     - Has `omitempty` on the REST API
     - Is unset for broadcast messages
     - Filled in to "me" on origination of our own message
     - Filled in from the batch within the DX callback

Reply:
Yep, this is a good point. In adding the new group checks, I noted that the group often isn't available during the DX callback, because it's created by the aggregator. I somehow convinced myself that the node might similarly not be available... but you're correct that message would never have gotten delivered by DX if the node hadn't been successfully created 💡

I think that will actually simplify the change a bit.